### PR TITLE
[flang] IEEE_GET_UNDERFLOW_MODE, IEEE_SET_UNDERFLOW_MODE

### PR DESCRIPTION
**Pull request by vdonaldson**
Implement IEEE_GET_UNDERFLOW_MODE and IEEE_SET_UNDERFLOW_MODE. Update IEEE_SUPPORT_UNDERFLOW_CONTROL to enable support for indvidual REAL kinds.

### CODE REVIEW

**Review by klausler — APPROVED** [review]
LGTM.

**Review by psteinfeld — APPROVED** [review]
All builds and tests correctly and looks good.

**Review by jeanPerier — APPROVED** [review]
Looks great, thanks Val!

File: flang/include/flang/Evaluate/target.h [hunk 1]
```diff
@@ -52,6 +52,11 @@ class TargetCharacteristics {
   }
   void set_areSubnormalsFlushedToZero(bool yes = true);
 
+  // Check if a given real kind, any real kind, or all real kinds have
+  // flushing control.
+  bool hasSubnormalFlushingControl(int kind, bool any = false) const;
+  void set_hasSubnormalFlushingControl(int kind, bool yes = true);
```
Comment (vdonaldson) [inline flang/include/flang/Evaluate/target.h:58]: I'm testing a change that splits this function into two overloads.

**Pull request merged**